src/analysis/modal-utils.ts

```diff
@@ -33,61 +39,136 @@ export interface IComputeConfig {
   persistentDiskType: DiskType;
 }
 
+export interface ExistingModalRuntimeConfig {
+  hasGpu: boolean;
+  autopauseThreshold: number | undefined;
+  computeConfig: IComputeConfig;
+  runtime?: NormalizedModalRuntime;
+  currentRuntimeDetails?: GetRuntimeItem;
+  persistentDisk?: ExistingModalPersistentDiskConfig;
+}
+
+export type DesiredModalRuntimeConfig = {
+  persistentDisk?: DesiredModalPersistentDiskConfig;
+} & Omit<ExistingModalRuntimeConfig, 'persistentDisk'>;
+
+export interface ExistingModalPersistentDiskConfig {
+  size: number;
+  diskType: GooglePdType;
+}
+// The desired diverges from the existing because the api request to make a runtime requires `GoogleDiskType`
+// However, we decorate the type to `GooglePdType` when we get it from the api
+export interface DesiredModalPersistentDiskConfig {
+  size: number;
+  diskType: GoogleDiskType;
+}
+
+export type NormalizedModalRuntime = {
+  cloudService: ComputeType | undefined;
+  toolDockerImage: string | undefined; // this is a url
+  tool?: ToolLabel;
+  persistentDiskAttached: boolean;
+  region: string;
+  jupyterUserScriptUri?: string;
+  gpuConfig?: any; // TODO
+  zone?: string;
+  machineType?: string;
+  bootDiskSize?: number | null;
+  diskSize?: number;
+  componentGatewayEnabled?: boolean;
+  masterMachineType?: string;
+  masterDiskSize?: number;
+  numberOfWorkers?: number;
+  numberOfPreemptibleWorkers?: number | null;
+  workerMachineType?: string | null | undefined;
+  workerDiskSize?: number | null | undefined;
+  timeoutInMinutes?: number | null;
+};
+
+// Auxiliary functions -- begin
+export const isGce = (runtimeType) => runtimeType === runtimeTypes.gceVm;
```

Comment:
these three isXxxx functions could be ijmproved to be type predicates that return "is TypeX... But from our chat, I assume that will have to wait until we do a 2nd wave of type love.

Reply:
yeah, since that will require nesting them in an additional field and thus functionally changing the modal itself, which I was able to avoid in this PR